src/Syncs/BloomFilter.cpp

```diff
@@ -0,0 +1,160 @@
+/* This code is part of the GenSync project developed at Boston University.  Please see the README for use and references. */
+
+//
+// Created by Anish Sinha on 12/4/23.
+//
+
+#include <GenSync/Syncs/BloomFilter.h>
+
+BloomFilter::BloomFilter() = default;
+BloomFilter::~BloomFilter() = default;
+
+BloomFilter::BloomFilter(size_t size, int nHash)
+{
+    this->bfSize = size;
+    this->bits.resize(size, 0);
+    this->numHashes = nHash;
+}
+
+BloomFilter::BloomFilter(size_t numExpElems, float falsePosProb)
+{
+    if(falsePosProb < 0 || falsePosProb > 1)
+    {
+        falsePosProb = 0.05;

```

Comment:
You should warn the user that you are changing the false positive probability ... maybe it's best to simply throw an exception?